File: 35. Search Insert Position/GenSol.js
// 35. Search Insert Position
// https://leetcode.com/problems/search-insert-position/

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function(nums, target) {
    let left = 0, right = nums.length
    // left should be the min possible value and right should ne the max possible value
    while (left < right) {
        let middle = left + Math.floor((right-left)/2)
        target > nums[middle] ? left = middle + 1 : right = middle
    }
    return left
};

File: 35. Search Insert Position/GenSol.py
'''
35. Search Insert Position

https://leetcode.com/problems/search-insert-position/description/
'''

class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        left = 0
        right = len(nums) - 1
         
        while left <= right:
            middle = (left + right) // 2

            if nums[middle] == target: return middle
            elif nums[middle] > target: right = middle - 1
            else: left = middle + 1

        return left
    
    